src/Microsoft.Framework.Runtime.Hosting/RuntimeHostBuilder.cs

```diff
@@ -1,28 +1,72 @@
+using NuGet.Frameworks;
+using NuGet.ProjectModel;
 
 namespace Microsoft.Framework.Runtime.Hosting
 {
     public class RuntimeHostBuilder
     {
-        public string ApplicationBaseDirectory { get; }
+        public IAssemblyLoaderContainer LoaderContainer { get; }
+        public IList<IDependencyProvider> DependencyProviders { get; }
+        public NuGetFramework TargetFramework { get; set; }
+        public Project Project { get; set; }
+        public LockFile LockFile { get; set;  }
 
-        public RuntimeHostBuilder(string applicationBaseDirectory)
+        public RuntimeHostBuilder(IAssemblyLoaderContainer loaderContainer)
         {
-            ApplicationBaseDirectory = applicationBaseDirectory;
+            LoaderContainer = loaderContainer;
+            DependencyProviders = new List<IDependencyProvider>();
         }
 
-        public RuntimeHost Build()
+        /// <summary>
+        /// Create a <see cref="RuntimeHostBuilder"/> for the project in the specified
+        /// <paramref name="projectDirectory"/>.
+        /// </summary>
+        /// <remarks>
+        /// This method will throw if the project.json file cannot be found in the
+        /// specified folder. If a project.lock.json file is present in the directory
+        /// it will be loaded. 
+        /// </remarks>
+        /// <param name="projectDirectory">The directory of the project to host</param>
+        public static RuntimeHostBuilder ForProjectDirectory(string projectDirectory, IApplicationEnvironment applicationEnvironment, IAssemblyLoaderContainer loaderContainer)
         {
+            var hostBuilder = new RuntimeHostBuilder(loaderContainer);
+
             // Load the Project
-            Project project = null;
+            hostBuilder.Project = ProjectReader.ReadProjectFile(projectDirectory);
 
-            PackageSpec packageSpec;
-            if (JsonPackageSpecReader.TryReadPackageSpec(ApplicationBaseDirectory, out packageSpec))
+            // Load the Lock File if present
+            if (ProjectReader.HasLockFile(projectDirectory))
             {
-                project = new Project(packageSpec);
+                hostBuilder.LockFile = ProjectReader.ReadLockFile(projectDirectory);
             }
 
-            return new RuntimeHost(this, project);
+            // Set the framework
+            hostBuilder.TargetFramework = NuGetFramework.Parse(applicationEnvironment.RuntimeFramework.FullName);
+
+            var projectResolver = new PackageSpecResolver(projectDirectory);
+            hostBuilder.DependencyProviders.Add(new PackageSpecReferenceDependencyProvider(projectResolver));
+            var referenceResolver = new FrameworkReferenceResolver();
+            hostBuilder.DependencyProviders.Add(new ReferenceAssemblyDependencyProvider(referenceResolver));
+
+            if (hostBuilder.LockFile != null)
+            {
+                hostBuilder.DependencyProviders.Add(new LockFileDependencyProvider(hostBuilder.LockFile));
```

Comment:
:+1: I haven't actually wired up the GAC resolver yet, but I'll put it at the end when I do